### PR TITLE
(DOCS-4023) IDEA Plugin

### PR DESCRIPTION
**Pull request by hestonhoffman**
Documentation for the IDEA plugin

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/developers/ide_integrations/idea/_index.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+The plugin features:
+- Top List: Identify methods that consume the most resources
+- Flame graph: visualize aggregated profile data
+- Find resource consumption broken down by method name and line number
+- Insights available from various staging and production environments
+
+## Requirements
+
+- **A Datadog account**: The plugins require a datadog account. If you're new to the Datadog suite of products, go the [Datadog website][4] for to learn more about Datadog's observability tools and sign up for a free trial.
```
Comment (jfree) [inline content/en/developers/ide_integrations/idea/_index.md:33]: `go the Datadog website for to learn` -> `go to the Datadog website to learn`

File: content/en/developers/ide_integrations/idea/_index.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+To configure which Profiling data is used for points of interest, see [Code synchronization](#code-synchronization).
+
+## Code synchronization
+
+Some Datadog plugin features require synchronization with a Profiling tab:
+- [Items of interest](#items-of-interest) correspond to whichever profile is selected.
+- The IDEA project tree view is annotated with the selected profiles metrics.
+  {{< img src="/developers/ide_integrations/idea/project-tree-view.png" alt="The project tree annotated with profile metrics from a profile tab" style="width:60%;" >}}
+
+To control which profiling tab contributes data to the IDEA project tree or editor highlights, you can pin a profile, or let IDEA sync data according to whichever plugin is active.
```
Comment (jfree) [inline content/en/developers/ide_integrations/idea/_index.md:146]: `whichever plugin is active` -> `whichever profiling tab is active`

File: content/en/developers/ide_integrations/idea/_index.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+
+To sync a profile, in the desired Profiling tab, click on the **Sync** and:
+- Select **Sync with** and choose the name of the desired profile to pin that profile.
+- Select **Sync with active profile** to allow IDEA to sync data with whichever profile tab is active.
+
+After you select a profile, a yellow dot appears next to the selected tab's title to indicate that it has focus.
+
+## Feedback
+
+User feedback is important to us. Let us know what you think about the plugin, or provide feedback on how on our [discussion forum][1], or send an e-mail to `team-ide-integration@datadoghq.com`.
```
Comment (jfree) [inline content/en/developers/ide_integrations/idea/_index.md:156]: `provide feedback on how ??` -> is there a phrase missing?

File: content/en/developers/ide_integrations/idea/_index.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+### Install the Datadog plugin
+
+1. Click **Plugins** and search for `Datadog`.
+1. Click **Install** to download and install the plugin in your IDE.
+1. If you receive a prompt notifying you that Datadog is a third-party plugin, click **Accept**.
+1. Click **Restart IDE** to restart IDEA.
+
+{{< img src="/developers/ide_integrations/idea/datadog-plugin.png" alt="The Datadog plugin" style="width:100%;" >}}
+
+### Add Datadog credentials
```
Comment (jfree) [inline content/en/developers/ide_integrations/idea/_index.md:47]: At the bottom of the page there is a link to `Find out more about API and application keys`.  Would it be useful to refer to it directly here?  Some developers may be new to Datadog and not be aware of what these keys are, so they might be stuck at this step.

File: content/en/developers/ide_integrations/idea/_index.md [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+
+The plugin features:
+- Top List: Identify methods that consume the most resources
+- Flame graph: visualize aggregated profile data
+- Find resource consumption broken down by method name and line number
+- Insights available from various staging and production environments
+
+## Requirements
+
+- **A Datadog account**: The plugins require a datadog account. If you're new to the Datadog suite of products, go to the [Datadog website][4] to learn more about Datadog's observability tools and sign up for a free trial.
```
Comment (jfree) [inline content/en/developers/ide_integrations/idea/_index.md:33]: `plugins require a datadog account.` -> `plugin requires a Datadog account.`

File: content/en/developers/ide_integrations/idea/_index.md [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+- **Continuous Profiling**: To display code-level insights, the plugin requires Continuous Profiling instrumented on your Java Services. For more information, see [Getting Started with the Continuous Profiler][3].
```
Comment (jfree) [inline content/en/developers/ide_integrations/idea/_index.md:34]: What do you think about adding this link to the `Further Reading` section at the bottom of the page as well?

File: content/en/developers/ide_integrations/idea/_index.md [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+
+## Using the plugin
+
+After you add a service to your project, right-click on the service and click **Open in Profiling** to open a Profiling tab for the service. A profiling tab displays data for only one service, but you can have multiple tabs open simultaneously.
+
+The Profiling tab shows Continuous Profiling information for the service in a selected environment, aggregated over a specific time frame. Available views are:
+- [Top list](#top-list): Displays a list of the most resource intensive methods for the current profiling measure.
+- [Flame graph](#flame-graph): A flame graph representing stack traces in the profiles.
+
+{{< img src="/developers/ide_integrations/idea/profiling-example.png" alt="An example of the Profiling tab for a service named product-recommendation" style="width:100%;" >}}
```
Comment (jfree) [inline content/en/developers/ide_integrations/idea/_index.md:74]: This image shows text `Flamegraph` and includes the code sync button that will be removed.  A more recent image is https://user-images.githubusercontent.com/1835893/195404809-53b81808-8252-4306-81eb-1274933ef8c5.png

File: content/en/developers/ide_integrations/idea/_index.md [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+Hovering over a method displays a tooltip with the following information:
+- The class name and method signature
+- The package name
+- The profiling metric value and percentage breakdown.
+
+Profiling samples include stack trace and line number information. Use the **Separate Flame Graph by** button to switch between separating frames by method or line number.
+
+{{< img src="/developers/ide_integrations/idea/separate-flamegraph-by.png" alt="Use the tooltip button to separate frames by method or line number" style="width:40%;" >}}
+
+## Items of interest
```
Comment (jfree) [inline content/en/developers/ide_integrations/idea/_index.md:131]: Perhaps change this to `Datadog Insights`?  It is the term we use in the popup list.

File: content/en/developers/ide_integrations/idea/_index.md [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+- link: "/account_management/api-app-keys/"
+  tag: "Documentation"
+  text: "Find out more about API and application keys."
+- link: "/getting_started/profiler/"
+  tag: "Documentation"
+  text: "Getting started with Continuous Profiler."
+---
+
+{{< beta-callout url="#" btn_hidden="true">}}
+  The Datadog plugin for IntelliJ IDEA is in Public Beta. It is intended for customers that use the <a href="https://docs.datadoghq.com/profiler/#pagetitle">Continuous Profiler</a> for their Java services. If the plugin stops working unexpectedly, check for plugin updates.
```
Comment (EWachnowezki) [inline content/en/developers/ide_integrations/idea/_index.md:16]: maybe we should add, and reach out via email team handle?

File: content/en/developers/ide_integrations/idea/_index.md [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+{{< /beta-callout >}}
+
+## Overview
+
+The Datadog plugin for IntelliJ IDEA helps improve software performance by providing meaningful code-level insights in the IDE based on real-time observability data. The plugin helps reduce latency and lower cloud costs by highlighting code lines that:
```
Comment (EWachnowezki) [inline content/en/developers/ide_integrations/idea/_index.md:21]:
```suggestion
The Datadog plugin for IntelliJ IDEA helps improve software performance by providing meaningful code-level insights in the IDE based on real-time observability data. The continuous profiler with this plugin helps you to reduce latency and lower cloud costs by highlighting code lines that:
```

File: content/en/developers/ide_integrations/idea/_index.md [hunk 11]
```diff
@@ -0,0 +1,155 @@
...
+- consume the most CPU
+- allocate the most memory
+- spend the most time on locks, disk I/O, socket I/O, and so on
+
+{{< img src="/developers/ide_integrations/idea/overview.png" alt="The Datadog tool window open in IDEA" style="width:100%;" >}}
+
+The plugin features:
+- Top List: Identify methods that consume the most resources
```
Comment (EWachnowezki) [inline content/en/developers/ide_integrations/idea/_index.md:29]: Add something along these lines:
Navigate directly from Top List, Flame graph to the respective code line

File: content/en/developers/ide_integrations/idea/_index.md [hunk 12]
```diff
@@ -0,0 +1,155 @@
...
+
+1. Click **Plugins** and search for `Datadog`.
+1. Click **Install** to download and install the plugin in your IDE.
+1. If you receive a prompt notifying you that Datadog is a third-party plugin, click **Accept**.
+1. Click **Restart IDE** to restart IDEA.
+
+{{< img src="/developers/ide_integrations/idea/datadog-plugin1.png" alt="The Datadog plugin" style="width:100%;" >}}
+
+### Add Datadog credentials
+
```
Comment (EWachnowezki) [inline content/en/developers/ide_integrations/idea/_index.md:51]: Do we want to add a small banner that this subject to change after the BETA?

File: content/en/developers/ide_integrations/idea/_index.md [hunk 13]
```diff
@@ -0,0 +1,155 @@
...
+- Hover over the icon to see more information.
+- Click the icon to open the top list Profiling tab or open Profiling in Datadog.
+  {{< img src="/developers/ide_integrations/idea/interest-options.png" alt="Click the Datadog icon to open the Profiling data in a tab or in Datadog" style="width:100%;" >}}
+
+The active Profiling tab also affects the IDEA project tree view, which is annotated with the selected profile's metrics:
+{{< img src="/developers/ide_integrations/idea/project-tree-view.png" alt="The project tree annotated with profile metrics from a profile tab" style="width:60%;" >}}
+
+## Feedback
+
+Let us know what you think about the plugin! Provide feedback on our [discussion forum][1], or send an e-mail to `team-ide-integration@datadoghq.com`.
```
Comment (EWachnowezki) [inline content/en/developers/ide_integrations/idea/_index.md:146]: maybe also open and reference a public slack channel?

File: content/en/developers/ide_integrations/idea/_index.md [hunk 14]
```diff
@@ -0,0 +1,155 @@
...
+{{< /beta-callout >}}
+
+## Overview
+
+The Datadog plugin for IntelliJ IDEA helps improve software performance by providing meaningful code-level insights in the IDE based on real-time observability data. The plugin helps reduce latency and lower cloud costs by highlighting code lines that:
+- consume the most CPU
+- allocate the most memory
+- spend the most time on locks, disk I/O, socket I/O, and so on
+
+{{< img src="/developers/ide_integrations/idea/overview.png" alt="The Datadog tool window open in IDEA" style="width:100%;" >}}
```
Comment (jfree) [inline content/en/developers/ide_integrations/idea/_index.md:26]: This screenshot shows the now-removed `Code Sync` button - here is an updated image: https://user-images.githubusercontent.com/1835893/195759179-7c9fe642-229d-477f-bb6d-0f3ac74fbfb2.png

File: content/en/developers/ide_integrations/idea/_index.md [hunk 15]
```diff
@@ -0,0 +1,155 @@
...
+
+## Top list
+
+The **Top List** sub-tab shows the methods that consume the most resources based on the aggregated profile data loaded from the Datadog servers. **Top List** is designed to show a summary of the methods that are most likely to be interesting from a resource consumption point of view.
+
+{{< img src="/developers/ide_integrations/idea/top-list.png" alt="The Top-list view" style="width:100%;" >}}
+
+Double-clicking an item in the list, or selecting **Jump to Source** item from the context menu, opens a source code editor showing where the method is defined.
+
+{{< img src="/developers/ide_integrations/idea/jump-to-source.png" alt="Select Jump to Source from the context menu" style="width:100%;" >}}
```
Comment (jfree) [inline content/en/developers/ide_integrations/idea/_index.md:94]: Given your earlier comment about the docs being heavy on screenshots, I suggest that this one could be dropped - the text description is sufficient in my opinion.

File: content/en/developers/ide_integrations/idea/_index.md [hunk 16]
```diff
@@ -0,0 +1,155 @@
...
+- Right-click on a method and select **Jump to Source** to go to the corresponding point in the source code.
+
+Hovering over a method displays a tooltip with the following information:
+- The class name and method signature
+- The package name
+- The profiling metric value and percentage breakdown.
+
+Profiling samples include stack trace and line number information. Use the **Separate Flame Graph by** button to switch between separating frames by method or line number.
+
+{{< img src="/developers/ide_integrations/idea/separate-flamegraph-by.png" alt="Use the tooltip button to separate frames by method or line number" style="width:40%;" >}}
```
Comment (jfree) [inline content/en/developers/ide_integrations/idea/_index.md:132]: This screenshot shows the now-removed `Code Sync` button.  An updated version is here: https://user-images.githubusercontent.com/1835893/195759992-67daa2b7-eeb1-40bf-bab7-21344fc2ff2a.png

File: content/en/developers/ide_integrations/idea/_index.md [hunk 17]
```diff
@@ -0,0 +1,155 @@
...
+- The environment in which the service is running
+- The time frame for the profile samples to be aggregated
+
+The available profiling types usually include options like **CPU Time** and **Allocated Memory**, but are determined by the platform and vary by language.
+
+## Top list
+
+The **Top List** sub-tab shows the methods that consume the most resources based on the aggregated profile data loaded from the Datadog servers. **Top List** is designed to show a summary of the methods that are most likely to be interesting from a resource consumption point of view.
+
+{{< img src="/developers/ide_integrations/idea/top-list.png" alt="The Top-list view" style="width:100%;" >}}
```
Comment (jfree) [inline content/en/developers/ide_integrations/idea/_index.md:90]: This screenshot shows the now-removed `Code Sync` button. An updated version is here: https://user-images.githubusercontent.com/1835893/195761932-d11edc25-bc55-45d9-92ff-d0c1a1efeb62.png

File: content/en/developers/ide_integrations/idea/_index.md [hunk 18]
```diff
@@ -0,0 +1,155 @@
...
+
+{{< img src="/developers/ide_integrations/idea/jump-to-source.png" alt="Select Jump to Source from the context menu" style="width:100%;" >}}
+
+To see a flame graph visualization of a method, select the **Search in Flame Graph** item from the context menu.
+
+### Call tree
+
+The call tree to the right of the method list shows the paths that lead to (and from) the selected method.
+
+{{< img src="/developers/ide_integrations/idea/call-tree.png" alt="The method call tree" style="width:100%;" >}}
```
Comment (jfree) [inline content/en/developers/ide_integrations/idea/_index.md:102]: This image shows the now-removed `Code Sync` button. An updated version is here: https://user-images.githubusercontent.com/1835893/195762411-9042341e-55de-4aec-af31-8e57caef0c13.png

File: content/en/developers/ide_integrations/idea/_index.md (hunk 12 above)
Comment (hestonhoffman) [reply content/en/developers/ide_integrations/idea/_index.md:51]: Could you clarify? What is subject to change?

File: content/en/developers/ide_integrations/idea/_index.md (hunk 13 above)
Comment (hestonhoffman) [reply content/en/developers/ide_integrations/idea/_index.md:146]: If someone creates a channel, I can add it the doc.

File: content/en/developers/ide_integrations/idea/_index.md (hunk 16 above)
Comment (hestonhoffman) [reply content/en/developers/ide_integrations/idea/_index.md:132]: I think you mean the screenshot above (flamegraph.png), I'll switch it out

File: content/en/developers/ide_integrations/idea/_index.md (hunk 9 above)
Comment (hestonhoffman) [reply content/en/developers/ide_integrations/idea/_index.md]: I've updated the banner language to make it a little less impersonal and linked to the feedback section below. That way, users can look at all the different feedback options in one place.

File: content/en/developers/ide_integrations/idea/_index.md [hunk 19]
```diff
@@ -0,0 +1,150 @@
...
+{{< img src="/developers/ide_integrations/idea/datadog-plugin1.png" alt="The Datadog plugin" style="width:100%;" >}}
+
+### Add Datadog credentials
+
+After installing the Datadog plugin and restarting IDEA, add your Datadog API and application keys to authenticate with Datadog:
+1. With a file or project open in IDEA, click the **Datadog** tool window.
+1. Click **Add your credentials...**.
+1. Enter your name, [API key and application key][5], and select a site.
+
+**Note**: For most users, one pair of keys is all that is required. If you're using a multi-org setup with multiple key pairs, check to ensure that the correct pair is active using the **Tools** -> **Datadog** section in **Preferences**.
```
Comment (apigirl) [inline content/en/developers/ide_integrations/idea/_index.md:57]: a link might be useful

Comment (hestonhoffman) [reply content/en/developers/ide_integrations/idea/_index.md:57]: This is inside the IDEA app, I can probably refine though 🤔

**Review by jfree — APPROVED** [review]
Great work, thanks! No further suggestions from me.

**Review by apigirl — APPROVED** [review]

**Pull request merged**